### ISSUE TITLE
Is there a way to read GPU memory usage

### ISSUE THREAD

**Issue by hungweitseng** [RadeonOpenCompute/ROCm#584]
Is there a way to read the USED GPU memory real-time or from the rcprof trace?

**Comment by jlgreathouse** [comment]
Internally, we're currently working on a patch to allow this information to be visible through rocm-smi. See [this issue](https://github.com/RadeonOpenCompute/ROC-smi/issues/42) and [this proposed patch](https://github.com/RadeonOpenCompute/ROC-smi/pull/43).

You can get this data out of the following sysfs file: `/sys/class/kfd/kfd/topology/nodes/N/mem_banks/0/used_memory` (and you can get available memory from `/sys/class/kfd/kfd/topology/nodes/N/mem_banks/0/properties` from the field `size_in_bytes`). In this case, `N` should likely be the ROCm agent ID (if you run `rocminfo` look at the Agent value and subtract 1).

**Comment by kentrussell** [comment]
The big issue that we have is that the topology/node/N is not a direct correlation to drm/cardN or dri/N . This is being worked on (see https://github.com/RadeonOpenCompute/ROC-smi/issues/42) but for now, you can use this command to look at it in real time until we get it implemented in the SMI:
sudo cat /sys/kernel/debug/dri/0/amdgpu_vram_mm | tail -n 2

Once we close the issue linked by Joe and I, the SMI will have support for it, but for now the easiest way for the GPU is to use my command here. Joe's command works well but can be problematic if you have supported CPU nodes that KFD uses, since they don't have used_memory. It's a complicated web

**Comment by hungweitseng** [comment]
Thanks. Will try this work-around first.

**Comment by kentrussell** [comment]
This will be available in the 2.3 release via the SMI

**Comment by jlgreathouse** [comment]
This has been available since ROCm 2.3 using `rocm-smi --showmeminfo vram`. I'll note that ROCm 2.3 and 2.4 reversed the "used" and "total" print-outs. The numbers are accurate, but the labels are reversed. This should be fixed in 2.5 I believe.

**Comment by jGsch** [comment]
Same error with rocm 3.7 (with docker image `rocm/pytorch:rocm3.7_ubuntu18.04_py3.6_pytorch_gcc`)

```
root@cn2:~# /opt/rocm/bin/rocm-smi

========================ROCm System Management Interface========================
================================================================================
GPU  Temp   AvgPwr  SCLK     MCLK    Fan     Perf  PwrCap  VRAM%  GPU%
0    60.0c  189.0W  1630Mhz  945Mhz  17.65%  auto  200.0W  N/A    98%
================================================================================
==============================End of ROCm SMI Log ==============================
```

About the used gpu:
```
root@cn2:~# /opt/rocm/bin/rocm_agent_enumerator
gfx000
gfx900
```
